projects/types.nix

```diff
@@ -231,6 +278,10 @@ let
                           type = nullOr (attrsOf (nullOr types'.service));
                           default = null;
                         };
+                        demo = lib.mkOption {
+                          type = nullOr (attrsOf types'.demo);
```

Comment:
Some design choices for `nixos.demo`:

1. attrTag of `vm` or `shell`. Only one can be set at a time.
1. `attrsOf types'.demo`, with restriction on kind (attr name should be either enum of `vm` or `shell`)
1. `types'.demo` with `kind` (vm or demo) -> means we'll only have 1 demo for a project, so we can't have multiple ones for service and shells

Reply:
AttrTag can only be one, and it allows clearly separate types for each tag. Always prefer that as a "structured enum" (aka sum type)